src/prefect/task_server.py

```diff
@@ -225,16 +216,22 @@ async def _submit_scheduled_task_run(self, task_run: TaskRun):
             validated_state=state,
         )
 
-        self._runs_task_group.start_soon(
-            partial(
-                submit_autonomous_task_run_to_engine,
+        if task.isasync:
```

Comment:
this works well, but as an FYI we also have a top-level `run_task` function that does this switching on your behalf if you'd prefer - up to you

Reply:
Oh nice!